File: 74. Search a 2D Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row=matrix.length;
        int col=matrix[0].length;
        boolean result=false;
        int m=0;
        int n=col-1;
        while(m<row && n>=0){
            if(matrix[m][n]==target){
                result=true;
                break;
            }
            if(matrix[m][n]<target){
                m++;
            }
            else{
                n--;
            }
        }
        return result;

    }
}

File: 74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row=len(matrix)
        col=len(matrix[0])
        i=0
        j=col-1
        result=False
        while i<row and j>=0:
            if matrix[i][j]==target:
                result=True
                break
            if matrix[i][j]>target:
                j-=1
            else:
                i+=1
        return result
